36/upgrade.html

```diff
@@ -214,10 +214,8 @@ <h5><a id="upgrade_350_kraft" href="#upgrade_350_kraft">Upgrading KRaft-based cl
                 ./bin/kafka-features.sh upgrade --metadata 3.5
             </code>
         </li>
-        <li>Note that cluster metadata downgrade is not supported in this version since it has metadata changes.
-            Every <a href="https://github.com/apache/kafka/blob/trunk/server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java">MetadataVersion</a>
-            after 3.2.x has a boolean parameter that indicates if there are metadata changes (i.e. <code>IBP_3_3_IV3(7, "3.3", "IV3", true)</code> means this version has metadata changes).
-            Given your current and target versions, a downgrade is only possible if there are no metadata changes in the versions between.</li>
+        <li>Note that the cluster metadata version cannot be downgraded to a pre-production 3.0.x, 3.1.x, or 3.2.x version once it has been upgraded.
```

Comment:
not sure about the source of truth here either

Reply:
ok so this was fixed in 3.7+ onwards for the old versions with https://github.com/apache/kafka/commit/aec07f76d763068feb6c1d19e4fc326cffd9c620